components/Modal/EditFlightModal.js

```diff
@@ -0,0 +1,104 @@
+import { useState } from "react";
+import { createPortal } from "react-dom";
+
+import {
+  ModalOverlay,
+  ModalContainer,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  CancelButton,
+  SubmitButton,
+  InputLabel,
+  InputField,
+} from "./EditFlightModal.styled";
+
+export default function EditFlightModal({ onClose, selectedFlight, onSave }) {
+  const [flightIata, setFlightIata] = useState(
+    selectedFlight?.flight_iata || ""
+  );
+  const [scheduledDate, setScheduledDate] = useState(
+    selectedFlight?.scheduled_date || ""
+  );
+  const [departure, setDeparture] = useState(selectedFlight?.departure || "");
+  const [arrival, setArrival] = useState(selectedFlight?.arrival || "");
+
+  const handleFlightIataChange = (event) => {
+    setFlightIata(event.target.value);
+  };
+  const handleScheduledDateChange = (event) => {
+    setScheduledDate(event.target.value);
+  };
+
+  const handleDepartureChange = (event) => {
+    setDeparture(event.target.value);
+  };
+
+  const handleArrivalChange = (event) => {
+    setArrival(event.target.value);
+  };
+
+  const handleSaveChanges = () => {
+    event.preventDefault();
+    const editedFlightData = {
+      flight_iata: flightIata,
+      scheduled_date: scheduledDate,
+      departure: departure,
+      arrival: arrival,
+    };
+
+    onSave(editedFlightData);
+    onClose();
+  };
+
+  const modalContent = (
+    <ModalOverlay onClose={onClose}>
+      <ModalContainer>
+        <form>
+          <ModalHeader>Edit Flight</ModalHeader>
```

Comment:
The submit event is triggered by the submit button, but it's caught by the form itself.
```suggestion
        <form onSubmit={handleSaveChanges}>
```

Reply:
I inverted the lines and added `onSubmit={handleSaveChanges}` to the form:
```
<ModalHeader>Edit Flight</ModalHeader>
<form onSubmit={handleSaveChanges}>
```